ethcore/src/executive.rs

```diff
@@ -245,30 +246,51 @@ impl<'a> Executive<'a> {
 		}
 		trace!("Executive::call(params={:?}) self.env_info={:?}", params, self.info);
 
+		let delegate_call = params.code_address != params.address;
```

Comment:
Ahh... you are right. But this edge case is not trivial to fix. With our current code design `Executive` does not know if `A` callcode `B` which callcode `A`, because it's irrelevant to `A` state (the result of this operation is the same as `A` calling itself).

I suggest leaving this as it is and fixing later, cause it may require additional changes in evm (exposing extra data).